core-plugins/src/main/java/io/cdap/plugin/batch/sink/TableSink.java

```diff
@@ -73,13 +73,19 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     Schema outputSchema =
       SchemaValidator.validateOutputSchemaAndInputSchemaIfPresent(tableSinkConfig.getSchemaStr(),
                                                                   tableSinkConfig.getRowField(), pipelineConfigurer);
-    if ((outputSchema != null) && (outputSchema.getFields().size() == 1)) {
-      String fieldName = outputSchema.getFields().get(0).getName();
-      if (fieldName.equals(tableSinkConfig.getRowField())) {
+    if (outputSchema != null) {
+      if (outputSchema.getFields().size() == 1) {
```

Comment:
hmmm this code looks suspicious. I think there are three different error scenarios:
a) schema has one field and that field is the row field
b) schema has one field and that field is different from the row field
c) schema has more fields but does not contain the row field

It looks as if case b) is not covered?

Reply:
Moved fieldName assignment out of conditional and merged logic from nested conditional (lmk if it makes more sense to just remove fieldName assignment and use outputSchema.getFields().get(0).getName().equals(tableSinkConfig.getRowField()).

Also, updated the tests. PTAL.

Reply:
It seems easier to read the code if: 
1. first check if the row key is in the schema
2. check if there are more than 1 field

Also, please add tests for each case. Thanks

Reply:
Yep! Added the extra test, as well.